rust/benches/car_benchmark.rs

```diff
@@ -0,0 +1,189 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use examples_uk_co_real_logic_sbe_benchmarks::*;
+
+const MANUFACTURER: &[u8] = b"MANUFACTURER";
+const MODEL: &[u8] = b"MODEL";
+
+struct State {
+    buffer: Vec<u8>,
+}
+
+pub fn car_encode_benchmark(c: &mut Criterion) {
+    let mut state = State {
+        buffer: vec![0u8; 1024],
+    };
+
+    c.bench_function("encode car", |b| {
+        b.iter(|| {
+            let _length = encode(black_box(&mut state)).unwrap();
+        })
+    });
+}
+
+fn encode(state: &mut State) -> SbeResult<usize> {
+    let buffer = state.buffer.as_mut_slice();
+    let mut car = CarEncoder::default();
+    let mut engine = EngineEncoder::default();
+    let mut fuel_figures = FuelFiguresEncoder::default();
+    let mut performance_figures = PerformanceFiguresEncoder::default();
+    let mut acceleration = AccelerationEncoder::default();
+    let mut extras = OptionalExtras::default();
+
+    car = car.wrap(WriteBuf::new(buffer), message_header::ENCODED_LENGTH);
+    car = car.header(0).parent()?;
+
+    car.code(Model::A);
+    car.model_year(2005);
+    car.serial_number(12345);
+    car.available(BooleanType::T);
+    car.vehicle_code([97, 98, 99, 100, 101, 102]); // abcdef
+    car.some_numbers([0, 1, 2, 3, 4]);
+
+    extras.set_sports_pack(true);
+    extras.set_sun_roof(true);
+    car.extras(extras);
+
+    engine = car.engine_encoder(engine);
+    engine.capacity(4200);
+    engine.num_cylinders(8);
+    engine.manufacturer_code([97, 98, 99]); // abc
+
+    car = engine.parent()?;
+    fuel_figures = car.fuel_figures_encoder(3, fuel_figures);
+    fuel_figures.advance()?;
+    fuel_figures.speed(30);
+    fuel_figures.mpg(35.9);
+
+    fuel_figures.advance()?;
+    fuel_figures.speed(55);
+    fuel_figures.mpg(49.0);
+
+    fuel_figures.advance()?;
+    fuel_figures.speed(75);
+    fuel_figures.mpg(40.0);
+
+    car = fuel_figures.parent()?;
```

Comment:
I notice that we reassigned the `car` variable from each child item after processing.  Is this a necessary step that users have to follow?

Reply:
Not necessarily, you could assign it to a new variable, or replace `car`:

```rust
let mut car = fuel_figures.parent()?;
```

Reply:
Also worth noting that if you choose a new variable name, the prior variable ‘car’ will no longer be accessible.